### PR TITLE
feat: ping message [AR-2594]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2594" title="AR-2594" target="_blank">StoryAR-2594</a>  Receiving Ping
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

In conversation screen we are not handling ping messages from web

### Causes (Optional)

Ping messages are showed as not supported messages

### Solutions

Handle ping messages as system messages with proper sound and separate notification channel

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/1321

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)
Screenshot 2023-01-11 at 14 12 31

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1317/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5449866)(5449866)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5449866)(5449866)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1317/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5454963)(5454963)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5454963)(5454963)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1317/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5455741)(5455741)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5455741)(5455741)-dev-compat.apk)

File: app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt [hunk 1]
```diff
@@ -157,7 +157,13 @@ class MessageContentMapper @Inject constructor(
             val assetMessageContentMetadata = AssetMessageContentMetadata(content.value)
             toUIMessageContent(assetMessageContentMetadata, message, sender)
         }
-
+        is MessageContent.Knock -> UIMessageContent.SystemMessage.Knock( // TODO should we move Knock to System message also in DAO layer?
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt:160]: On collective we discussed that and decided that ping should be considered as a regular message but should look like a system message, I think we can remove this TODO 😉

File: app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt [hunk 2]
```diff
@@ -32,16 +32,18 @@ class MessageNotificationManager
...
         if (newNotifications.isEmpty()) return
 
         val activeNotifications: Array<StatusBarNotification> = notificationManager.activeNotifications ?: arrayOf()
 
-        showSummaryIfNeeded(userId, activeNotifications, userName)
         newNotifications.forEach {
             showConversationNotification(it.intoNotificationConversation(), userId, activeNotifications)
         }
 
+        showSummaryIfNeeded(userId, activeNotifications, userName)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt:45]: Why is the order changed? 🤔

File: app/src/main/kotlin/com/wire/android/notification/NotificationChannelsManager.kt [hunk 3]
```diff
@@ -114,6 +121,22 @@ class NotificationChannelsManager @Inject constructor(
...
+        val audioAttributes = AudioAttributes.Builder()
+            .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+            .setUsage(AudioAttributes.USAGE_NOTIFICATION)
+            .build()
+
+        val notificationChannel = NotificationChannelCompat
+            .Builder(NotificationConstants.getPingsChannelId(userId), NotificationManagerCompat.IMPORTANCE_HIGH)
+            .setName(NotificationConstants.PING_CHANNEL_NAME)
+            .setGroup(channelGroupId)
+            .setSound(knockSoundUri, audioAttributes)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/notification/NotificationChannelsManager.kt:134]: So the sound of ping is played by the notification, right? What if we have the conversation screen open? Then we don't show notifications for that specific conversation - how can the user hear the ping then?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 4]
```diff
@@ -76,7 +75,7 @@ fun MessageItem(
                 .fillMaxWidth()
                 .let {
                     if (!message.isDeleted) it.combinedClickable(
-                        //TODO: implement some action onClick
+                        // TODO: implement some action onClick
                         onClick = { },
                         onLongClick = { onLongClicked(message) }
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:80]: Ping message types shouldn't be long clicked, right? Should we block it here or in the place where we open the message options bottom sheet? If the second solution then we should also remove the ripple when long clicking

File: app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt:160]: I will refactor It after meeting with PM and Designer

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1317/5/) **failed**.

File: app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt:45]: There was a problem with sound play order which caused that sometimes ping messages had normal notification sound. It was discussed with @borichellow

File: app/src/main/kotlin/com/wire/android/notification/NotificationChannelsManager.kt (hunk 3 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/notification/NotificationChannelsManager.kt:134]: I will ask about it and do improvements in next PR

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 4 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:80]: For now Ping is a System Message so long click is not implemented for it

**Review by saleniuk — APPROVED** [review]
Ok, but remember to make improvements to sound and message type in the next PR 😄

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1317/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5465348)(5465348)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5465348)(5465348)-dev-compat.apk)

**Pull request merged**